src/ui/public/courier/fetch/__tests__/call_client.js

```diff
@@ -0,0 +1,270 @@
+      const searchRequest = createSearchRequest(1, {
+        source: {
+          _flatten: () => { throw new Error(); },
+        },
+      });
+
+      searchRequests = [ searchRequest ];
+      const callingClient = callClient(searchRequests);
+
+      callingClient.then(results => {
+        // The result is ABORTED because it was never included in the body sent to es.msearch().
+        expect(results).to.eql([ ABORTED ]);
+        done();
+      }).catch(error => done(error));
+    });
+
+    it('while the search body is being formed resolves with an ABORTED response', done => {
+      const searchRequest = createSearchRequest(1, {
+        source: {
+          _flatten: () => {
+            return new Promise(resolve => {
+              setTimeout(() => {
+                resolve({});
+              }, 100);
+            });
+          },
+          requestIsStopped: () => {},
+        },
+      });
+
+      searchRequests = [ searchRequest ];
+      const callingClient = callClient(searchRequests);
+
+      // Abort the request while the search body is being formed.
+      setTimeout(() => {
+        searchRequest.abort();
+      }, 20);
+
+      callingClient.then(results => {
+        expect(results).to.eql([ ABORTED ]);
+        done();
+      }).catch(error => done(error));
+    });
+
+    it('while the search is in flight resolves with an ABORTED response', done => {
+      esRequestDelay = 100;
+
+      const searchRequest = createSearchRequest();
+      searchRequests = [ searchRequest ];
+      const callingClient = callClient(searchRequests);
+
+      // Abort the request while the search is in flight..
+      setTimeout(() => {
+        searchRequest.abort();
+      }, 80);
+
+      callingClient.then(results => {
+        expect(results).to.eql([ ABORTED ]);
+        done();
+      }).catch(error => done(error));
+    });
+  });
+
+  describe('aborting number of requests:', () => {
+    it(`aborting all searchRequests resolves with ABORTED responses`, done => {
+      const searchRequest1 = createSearchRequest();
+      const searchRequest2 = createSearchRequest();
+      searchRequests = [ searchRequest1, searchRequest2 ];
+      const callingClient = callClient(searchRequests);
+
+      searchRequest1.abort();
+      searchRequest2.abort();
+
+      callingClient.then(results => {
+        expect(results).to.eql([ABORTED, ABORTED]);
+        done();
+      }).catch(error => done(error));
+    });
+
+    it(`aborting all searchRequests calls abort() on the promise returned by es.msearch()`, done => {
+      esRequestDelay = 100;
+
+      const searchRequest1 = createSearchRequest();
+      const searchRequest2 = createSearchRequest();
+      searchRequests = [ searchRequest1, searchRequest2 ];
+
+      const callingClient = callClient(searchRequests);
+
+      setTimeout(() => {
+        // At this point we expect the request to be in flight.
+        expect(esPromiseAbortSpy.callCount).to.be(0);
```

Comment:
If `esPromiseAbortSpy` was called before `searchRequest1.abort()` is called then this will throw an error, which won't be caught or funneled to mocha in any way because the callback interface is in use instead of promises. On top of that, the `esPromiseAbortSpy.callCount` could still very possibly be `1` below causing the test to pass. I would organize the test like this instead:


```js
return Promise.all(
  delay(70).then(() => {
)
```